numpy/f2py/tests/src/parameter/constant_array.f90

```diff
@@ -2,14 +2,16 @@
 subroutine foo_array(x, y, z)
   implicit none
   integer, parameter :: dp = selected_real_kind(15)
+  integer, parameter :: pa = 2
   integer, parameter :: intparamarray(2) = (/ 3, 5 /)
+  integer, parameter, dimension(pa) :: pb = (/ 2, 10 /)
```

Comment:
Is this legal fortran?
```
integer, parameter, dimension(intparamarray(1)) :: pc = (/ 2, 10, 20 /)
```
I think this case currently fails.

Reply:
I have added this test and it does not fail. In my understanding, as long as the dimension attribute is a constant expression this is legal under the language specification. However if you feel like this should be investigated further I can check other compilers (my own tests are only run using gfortran for the moment).